server/jetstream_api.go

```diff
@@ -1189,19 +1245,65 @@ func (s *Server) jsStreamSnapshotRequest(sub *subscription, c *client, subject,
 	// We will do the snapshot in a go routine as well since check msgs may
 	// stall this go routine.
 	go func() {
+		s.Noticef("Starting snapshot for stream %q in account %q", mset.Name(), mset.jsa.account.Name)
+		start := time.Now()
+
 		sr, err := mset.Snapshot(0, !req.NoConsumers, req.CheckMsgs)
 		if err != nil {
+			s.Noticef("Snapshot of %q in account %q failed: %s", mset.Name(), mset.jsa.account.Name, err)
 			resp.Error = jsError(err)
 			s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 			return
 		}
 
 		resp.NumBlks = sr.NumBlks
 		resp.BlkSize = sr.BlkSize
+
 		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(resp))
 
+		sad := JSSnapshotCreateAdvisory{
+			TypedEvent: TypedEvent{
+				Type: JSSnapshotCreatedAdvisoryType,
+				ID:   nuid.Next(),
+				Time: time.Now().UTC(),
+			},
+			Stream:  mset.Name(),
+			NumBlks: sr.NumBlks,
+			BlkSize: sr.BlkSize,
+			Client:  c.apiAuditClient(),
+		}
+
+		ej, err := json.MarshalIndent(sad, "", "  ")
+		if err == nil {
+			s.sendInternalAccountMsg(c.acc, JSAdvisoryStreamSnapshotCreatePre+"."+mset.Name(), ej)
+		} else {
+			s.Warnf("JetStream could not marshal snapshot event for account %q: %v", c.acc.Name, err)
+		}
+
 		// Now do the real streaming.
 		s.streamSnapshot(c, mset, sr, &req)
+
+		end := time.Now()
+
+		cad := JSSnapshotCompleteAdvisory{
+			TypedEvent: TypedEvent{
+				Type: JSSnapshotCompleteAdvisoryType,
+				ID:   nuid.Next(),
+				Time: time.Now().UTC(),
+			},
+			Stream: mset.Name(),
+			Start:  start.UTC(),
+			End:    end.UTC(),
+			Client: sad.Client,
+		}
+		ej, err = json.MarshalIndent(cad, "", "  ")
```

Comment:
We should make advisory helper that takes all advisories and does json in the helper.

Reply:
Yup I wanted to. Will see.